library/alloc/src/collections/btree/node.rs

```diff
@@ -512,6 +525,24 @@ impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
     }
 }
 
+impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::ValMut<'a>, K, V, Type> {
```

Comment:
This hands out a `&'a mut [K]` to the key slice despite being just ValMut, which seems wrong?

Reply:
Yes, but this PR tries to only stick the ValMut label on (copies of) existing code and #73971 replaces this copy of `into_slices_mut` entirely. Though I did already remove the `mut`-ness of tbe keys in `into_kv_valmut`, which also gets replaced, so that's not consistent. And in the navigate methods and all the way up to the map functions, but these take on their final form.

I guess, looking at this in solation, it's not sound to call it ValMut and have `mut` keys. One could also argue that changing keys is not a problem, as long as you preserve order. Like we used to manually renumber lines in Basic programs, What really matters in this PR is the separation between structure altering handle types and methods versus K/V-altering tpyes and methods. So instead of ValMut, it could be KeyValMut, or ImmutLayout  or so.

Reply:
Anyways, fixed according to the ValMut philosophy.

Reply:
Yeah I see that now, but I think if we're going to call it ValMut and not something like TreeImmut (which sounds confusing, don't use it), we should go with this.